name: Conformance Tests

on:
  push:
    branches:
      - main
  workflow_dispatch:
  pull_request:

jobs:
  conformance:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2

      - uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
        with:
          python-version: "3.x"
          cache: "pip"
          cache-dependency-path: pyproject.toml

      - name: install sigstore-python
        run: python -m pip install .

      # TODO: Re-pin this to a tag.
      - uses: sigstore/sigstore-conformance@0cedff293ec007b0e67f7954b5dc044f5012f107
        with:
          entrypoint: ${{ github.workspace }}/test/integration/sigstore-python-conformance
          xfail: "test_verify_with_trust_root test_verify_dsse_bundle_with_trust_root" # see issue 821
